lib/puppet/indirector/resource_type/parser.rb

```diff
@@ -100,4 +100,9 @@ def resource_types_in(environment)
     environment.check_for_reparse
     environment.known_resource_types
   end
+
+  def allow_remote_requests?
+    Puppet.deprecation_warning("The resource_type endpoint is deprecated in favor of the environment_classes endpoint. See https://github.com/puppetlabs/puppet-server/blob/master/documentation/puppet-api/v3/environment_classes.md")
```

Comment:
A pretty common pattern is to register a links.puppetlabs.com link. Unfortunately that was broken in the move to puppet.com. Follow along (internal-only sorry) at https://tickets.puppetlabs.com/browse/OPS-8714.